plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraTableHandle.java

```diff
@@ -14,79 +14,107 @@
 import io.trino.spi.connector.SchemaTableName;
 
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
-import java.util.stream.Stream;
 
-import static java.lang.String.format;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
-import static java.util.stream.Collectors.joining;
 
 public class CassandraTableHandle
         implements ConnectorTableHandle
 {
-    private final String schemaName;
-    private final String tableName;
-    private final Optional<List<CassandraPartition>> partitions;
-    private final String clusteringKeyPredicates;
+    private final CassandraRelationHandle relationHandle;
 
+    @Deprecated
     public CassandraTableHandle(String schemaName, String tableName)
     {
         this(schemaName, tableName, Optional.empty(), "");
     }
 
+    @Deprecated
+    public CassandraTableHandle(String schemaName, String tableName, Optional<List<CassandraPartition>> partitions, String clusteringKeyPredicates)
+    {
+        this(new CassandraNamedRelationHandle(schemaName, tableName, partitions, clusteringKeyPredicates));
+    }
+
     @JsonCreator
-    public CassandraTableHandle(
-            @JsonProperty("schemaName") String schemaName,
-            @JsonProperty("tableName") String tableName,
-            @JsonProperty("partitions") Optional<List<CassandraPartition>> partitions,
-            @JsonProperty("clusteringKeyPredicates") String clusteringKeyPredicates)
+    public CassandraTableHandle(@JsonProperty("relationHandle") CassandraRelationHandle relationHandle)
     {
-        this.schemaName = requireNonNull(schemaName, "schemaName is null");
-        this.tableName = requireNonNull(tableName, "tableName is null");
-        this.partitions = partitions.map(ImmutableList::copyOf);
-        this.clusteringKeyPredicates = requireNonNull(clusteringKeyPredicates, "clusteringKeyPredicates is null");
+        this.relationHandle = requireNonNull(relationHandle, "relationHandle is null");
     }
 
     @JsonProperty
+    public CassandraRelationHandle getRelationHandle()
+    {
+        return relationHandle;
+    }
+
+    @JsonIgnore
+    public CassandraNamedRelationHandle getRequiredNamedRelation()
+    {
+        checkState(isNamedRelation(), "The table handle does not represent a named relation: %s", this);
+        return (CassandraNamedRelationHandle) relationHandle;
+    }
+
+    @JsonIgnore
+    public boolean isSynthetic()
+    {
+        return !isNamedRelation();
+    }
+
+    @JsonIgnore
+    public boolean isNamedRelation()
+    {
+        return relationHandle instanceof CassandraNamedRelationHandle;
+    }
+
+    @JsonIgnore
+    public CassandraNamedRelationHandle asPlainTable()
+    {
+        checkState(!isSynthetic(), "The table handle does not represent a plain table: %s", this);
+        return getRequiredNamedRelation();
```

Comment:
Isn't this API a bit similar to `getRequiredNamedRelation` or can we have one copy of it.